docs_user/modules/proc_deploying-a-ceph-ingress-daemon.adoc

```diff
@@ -0,0 +1,131 @@
+
+.Procedure
+
+. Log in to each Controller node and remove the following configuration from the `/var/lib/config-data/puppet-generated/haproxy/etc/haproxy/haproxy.cfg` file:
++
+----
+listen ceph_rgw
+  bind 10.0.0.103:8080 transparent
+  mode http
+  balance leastconn
+  http-request set-header X-Forwarded-Proto https if { ssl_fc }
+  http-request set-header X-Forwarded-Proto http if !{ ssl_fc }
+  http-request set-header X-Forwarded-Port %[dst_port]
+  option httpchk GET /swift/healthcheck
+  option httplog
+  option forwardfor
+   server controller-0.storage.redhat.local 172.17.3.73:8080 check fall 5 inter 2000 rise 2
+  server controller-1.storage.redhat.local 172.17.3.146:8080 check fall 5 inter 2000 rise 2
+  server controller-2.storage.redhat.local 172.17.3.156:8080 check fall 5 inter 2000 rise 2
+----
+
+. Restart `haproxy-bundle` and ensure it is started:
++
+----
+[root@controller-0 ~]# sudo pcs resource restart haproxy-bundle
+haproxy-bundle successfully restarted
+
+
+[root@controller-0 ~]# sudo pcs status | grep haproxy
+
+  * Container bundle set: haproxy-bundle [undercloud-0.ctlplane.redhat.local:8787/rh-osbs/rhosp17-openstack-haproxy:pcmklatest]:
+    * haproxy-bundle-podman-0   (ocf:heartbeat:podman):  Started controller-0
+    * haproxy-bundle-podman-1   (ocf:heartbeat:podman):  Started controller-1
+    * haproxy-bundle-podman-2   (ocf:heartbeat:podman):  Started controller-2
+----
+
+. Confirm that no process is bound to 8080:
++
+----
+[root@controller-0 ~]# ss -antop | grep 8080
+[root@controller-0 ~]#
+----
++
+The swift CLI fails at this point:
++
+----
+(overcloud) [root@cephstorage-0 ~]# swift list
+
+HTTPConnectionPool(host='10.0.0.103', port=8080): Max retries exceeded with url: /swift/v1/AUTH_852f24425bb54fa896476af48cbe35d3?format=json (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fc41beb0430>: Failed to establish a new connection: [Errno 111] Connection refused'))
+----
+
+. Set the required images for both HAProxy and Keepalived: 
++ 
+----
+ifeval::["{build}" != "downstream"]
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_haproxy quay.io/ceph/haproxy:2.3
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_keepalived quay.io/ceph/keepalived:2.1.5
+endif::[]
+ifeval::["{build}" == "downstream"]
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_haproxy registry.redhat.io/rhceph/rhceph-haproxy-rhel9:latest
+[ceph: root@controller-0 /]# ceph config set mgr mgr/cephadm/container_image_keepalived registry.redhat.io/rhceph/keepalived-rhel9:latest
+endif::[]
+----
+
+. Prepare the ingress spec and mount it to cephadm:
++
+----
+$ sudo vim /home/ceph-admin/specs/rgw_ingress
+----
+
+. Copy and paste the following content in to the ingress spec:
+//kgilliga: Is this accurate? The user is pasting the content in to the ingress spec?
```

Comment:
feel free to rewrite  this part and match the required style.
The human operator should:
-  create a file called `rgw_ingress` (to keep things simple we put it in `/home/ceph-admin/specs` in the controller-0: that path already exists for a director deployed ceph)
- paste the following content to the file:

```
---
service_type: ingress
service_id: rgw.rgw
placement:
  label: rgw
spec:
  backend_service: rgw.rgw
  virtual_ip: 10.0.0.89/24
  frontend_port: 8080
  monitor_port: 8898
  virtual_interface_networks:
    - 10.0.0.0/24
```
Note: Replace eventually 10.0.0.0/24 to match the external network identified earlier

- Apply the new spec using the ceph orchestrator:

```
$ cephadm shell -m /home/ceph-admin/specs/rgw_ingress --  ceph orch apply -i /mnt/rgw_ingress
```
...
...